### PR TITLE
Add decorator pattern

### PR DESCRIPTION
**Pull request by mikemtzp**
- Create a class `Nameable.`
  - Implement a method called `correct_name` that will raise a `NotImplementedError.`

- Turn your Person class to Nameable
  - Make sure that your Person class inherits from Nameable
  - Make sure that this class has a method `correct_name` implemented. It should simply return the name attribute.

- Prepare base Decorator
  - Make sure that it inherits from Nameable.
  - In the constructor assign a nameable object from params to an instance variable.
  - Implement the correct_name method that returns the result of the `correct_name` method of the `@nameable.`

- Prepare `CapitalizeDecorator` and `TrimmerDecorator`:
  - For the CapitalizeDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method correct_name that capitalizes the output of @nameable.correct_name.
  - For the TrimmerDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: capitalize_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require './base_decorator'
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.upcase
```
Comment (zieeco) [inline capitalize_decorator.rb]: - Please, kindly ensure to capitalize the first letter of the given name. use `capitalize` to fix this issue.

File: trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require './base_decorator'
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name[0, 10]
```
Comment (zieeco) [inline trimmer_decorator.rb:5]: - Please, do ensure the name has a maximum of 10 characters. If it's longer it should trim the word, as of now, yours is 9 characters. please, fix this.

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @mikemtzp,


Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice implementation of the classes. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: trimmer_decorator.rb (hunk 2 above)
Comment (mikemtzp) [reply trimmer_decorator.rb:5]: @zieeco About this, I think the current code is working correctly as it reduces the name by a maximum of 10 characters. It reduces 'Maximilianus' with 12 characters to 'Maximilian' with 10 characters.😀 

image

File: capitalize_decorator.rb (hunk 1 above)
Comment (mikemtzp) [reply capitalize_decorator.rb]: Hi, @zieeco!
Thanks for the feedback! I have already modified the method by replacing `upcase()` with `capitalize().` 😁

**Review by rotimiazeez — APPROVED** [review]
# Hi @mikemtzp 👊🏽, 
## Good job making the requested changes 👍🏾
### Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**